tests/components/mazda/test_init.py

```diff
@@ -181,3 +183,130 @@ async def test_device_no_nickname(hass):
     assert reg_device.model == "2021 MAZDA3 2.5 S SE AWD"
     assert reg_device.manufacturer == "Mazda"
     assert reg_device.name == "2021 MAZDA3 2.5 S SE AWD"
+
+
+async def test_services(hass):
+    """Test service calls."""
+    client_mock = await init_integration(hass)
+
+    device_registry = dr.async_get(hass)
+    reg_device = device_registry.async_get_device(
+        identifiers={(DOMAIN, "JM000000000000000")},
+    )
+    device_id = reg_device.id
+
+    for service in SERVICES:
+        service_data = {"vehicle": device_id}
+        if service == "send_poi":
+            service_data["latitude"] = 1.2345
+            service_data["longitude"] = 2.3456
+            service_data["poi_name"] = "Work"
+
+        await hass.services.async_call(DOMAIN, service, service_data, blocking=True)
+        await hass.async_block_till_done()
+
+        api_method = getattr(client_mock, service)
+        if service == "send_poi":
```

Comment:
Please don't use conditional logic in the tests. Use `pytest.mark.parametrize` instead to test different cases.